src/lib/sys/bm_config_wrapper.cpp

```diff
@@ -86,11 +88,16 @@ bool bcmp_get_config(const char *key, size_t key_len, uint8_t *value, size_t &va
   } else {
     return false;
   }
-  return cfg->getConfigCbor(key, key_len, value, value_len);
+  return cfg->getConfigCbor(key, key_len, value, *value_len);
 }
 
 bool bm_cbor_type_to_config_type(const CborValue *value, GenericConfigDataTypes *configType) {
-  ConfigDataTypes_e return_config_type = (ConfigDataTypes_e)*configType;
+  bool ret = false;
+  ConfigDataTypes_e return_config_type;
   // TODO - have configuration.h use GenericConfigDataTypes
-  return Configuration::cborTypeToConfigType(value, return_config_type);
+  ret = Configuration::cborTypeToConfigType(value, return_config_type);
+  if (ret) {
+    *configType = (GenericConfigDataTypes)return_config_type;
+  }
+  return ret;
```

Comment:
IMPORTANT! THis logic had to change because the `bm cfg get` command was not working. We were not properly setting the `configType` passed in as reference here.